### ISSUE TITLE
Add Allowed_Sizes Option

### ISSUE THREAD

**Issue by lucasdss** [DarthSim/imgproxy#106]
For our scenario would be great to have an option to limit the resize for a few sizes.

Usually, sites need a bunch of pre-defined sizes like 1500x1500, 750x750, 375x375, 200x200, 100x100. And adding this restriction would be a good way to avoid so different images.

**Comment by DarthSim** [comment]
Since imgproxy URLs can be secured with a signature, I don't see any need to restrict any processing options. Could you explain the case where such a limitation could make sense?

**Comment by lucasdss** [comment]
I was thinking in this options in conjunction with the domain prefix. If some want to use imgproxy without creating all URL signatures previously it adds a reasonable use.

**Comment by DarthSim** [comment]
There is a large attack surface even with limited sizes. It's highly recommended to sign the URLs in production.

**Comment by lucasdss** [comment]
Agree. But what do you think if the size was removed from signature? Would make the use of signature easier and with this option, you still limit the attack with a really low cost.

**Comment by DarthSim** [comment]
This will make use of signature even harder as I will have to exclude size-related params from the path before signature checking, the user (administrator) will have to do the same. imgproxy itself at the same time will lose its flexibility.
